src/SourceBuild/content/Directory.Build.props

```diff
@@ -1,231 +1,196 @@
 <Project>
 
   <PropertyGroup>
-    <ImportNetSdkFromRepoToolset>false</ImportNetSdkFromRepoToolset>
-    <_SuppressSdkImports>true</_SuppressSdkImports>
+    <_SuppressSdkImports Condition="'$(DotNetBuildFromSource)' == 'true'">true</_SuppressSdkImports>
     <Configuration Condition="$(Configuration) == ''">Release</Configuration>
-    <NetCurrent Condition="'$(SkipArcadeSdkImport)' == 'true'">net9.0</NetCurrent>
   </PropertyGroup>
 
-  <Import Condition="'$(SkipArcadeSdkImport)' != 'true'" Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
-  <PropertyGroup>
-    <!-- Fake, for SDK. -->
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <!-- We have no projects targeting multiple frameworks, so don't include in output path. -->
-    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
-  </PropertyGroup>
+  <PropertyGroup Label="CalculateTargetOS">
+    <BuildOS>linux</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">osx</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('FREEBSD'))">freebsd</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('NETBSD'))">netbsd</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('ILLUMOS'))">illumos</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('SOLARIS'))">solaris</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('HAIKU'))">haiku</BuildOS>
+    <BuildOS Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">windows</BuildOS>
+    <HostOS Condition="'$(HostOS)' == ''">$(BuildOS)</HostOS>
+    <TargetOS Condition="'$(TargetOS)' == ''">$(BuildOS)</TargetOS>
+  </PropertyGroup>
+
+  <PropertyGroup Label="CalculateArch">
+    <BuildArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</BuildArchitecture>
+    <HostArchitecture Condition="'$(HostArchitecture)' == ''">$(BuildArchitecture)</HostArchitecture>
+    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' and '$(BuildArchitecture)' == 'arm'">arm</TargetArchitecture>
```

Comment:
Can you add a comment here re: the purpose of this? I think it's probably something like:
"When building on non-x64 architectures, there may be no cross-compilation available, select a target architecture that is the same as the build.

Reply:
This comes from runtime: https://github.com/dotnet/runtime/blob/1ad2ac015a9ef14dc49fb3075f48b1824a753486/Directory.Build.props#L38

Exactly. Will add the comment.